applications/ContactStructuralMechanicsApplication/symbolic_generation/ALM_frictionless_components_mortar_condition/generate_frictionless_components_mortar_condition.py

```diff
@@ -174,11 +174,11 @@ def convert_active_inactive_int(list_active):
                 active = active_inactive[node]
                 if (active == 1):
                     augmented_lm = (ScaleFactor * LM.row(node) + PenaltyParameter[node] * NormalGap[node] * NormalSlave.row(node))
-                    rv_galerkin += DynamicFactor[node] *  (augmented_lm).dot(Dw1Mw2.row(node))
-                    rv_galerkin += ScaleFactor * NormalGap[node] * wLMNormal[node]
-                    rv_galerkin += - ScaleFactor**2.0/PenaltyParameter[node] * (wLMTangent.row(node) * LMTangent.row(node).transpose())[0,0]
+                    rv_galerkin += DynamicFactor[node] * (augmented_lm).dot(Dw1Mw2.row(node))
+                    rv_galerkin -= ScaleFactor * NormalGap[node] * wLMNormal[node]
+                    rv_galerkin -= ScaleFactor**2.0/PenaltyParameter[node] * (wLMTangent.row(node).dot(LMTangent.row(node)))
```

Comment:
do you want the power to be a double??

Reply:
Yes, it is a double indeed

Reply:
can u explain me why... 

     ScaleFactor ** 2

should be the same as

    ScaleFactor**2.0

with the difference that such second one uses a very generic algorithm for computing the power and does not translate to ScaleFactor*ScaleFactor

Reply:
just as a "proof" of this try from the command line

>>> (-1)**2
1
>>> (-1)**2.0
1.0

the second one is a double!!